[CON-1589] feat: Extend Gitlab supported objects (#306)
amp-labs/docs: src/provider-guides/gitlab.mdx

---

> title: "Gitlab"

Reviewer: ```suggestion
title: "GitLab"
```

- [service_accounts](https://docs.gitlab.com/api/service_accounts/)
- [snippets](https://docs.gitlab.com/api/snippets)
- [todos/mark_as_done](https://docs.gitlab.com/api/todos/#mark-all-to-do-items-as-done)
- [topics](https://docs.gitlab.com/api/topics)
- [user/emails](https://docs.gitlab.com/api/user_email_addresses/#add-an-email-address)
- [user/gpg_keys](https://docs.gitlab.com/api/user/gpg_keys)
- [user/keys](https://docs.gitlab.com/api/user/keys)
- [user/runners](https://docs.gitlab.com/api/user/runners)
- [user/support_pin](https://docs.gitlab.com/api/users/#create-a-support-pin)
- [users](https://docs.gitlab.com/api/users)

### Example integration

For an example manifest file of an Gitlab integration, visit our [samples repo on Github](https://github.com/amp-labs/samples/blob/main/gitlab/amp.yaml).

## Before You Get Started

### Creating Access Token for Gitlab

[Click here](https://gitlab.com/-/user_settings/personal_access_tokens) for more information about generating an Access Token key for Gitlab. The UI components will display this link, so that your users can successfully create Access tokens.

## Using the connector

> This connector uses Access Token, which means that you do not need to set up a Provider App before getting started. (Provider apps are only required for providers that use OAuth2 Authorization Code grant type.)

Reviewer: ```suggestion
This connector uses GitLab Access Tokens for authentication, which means that you do not need to set up a Provider App before getting started. (Provider apps are only required for providers that use OAuth2 Authorization Code grant type.)

The access tokens can be:
- [Personal access tokens](https://docs.gitlab.com/user/profile/personal_access_tokens/)
- [Project access tokens](https://docs.gitlab.com/user/project/settings/project_access_tokens/)
- [Group access tokens](https://docs.gitlab.com/user/group/settings/group_access_tokens/)
```

> To start integrating with Apollo:

Reviewer: ```suggestion
To start integrating with Gitlab:
```
